alica_tests/src/test/test_blackboard.cpp

```diff
@@ -50,18 +52,99 @@ class TestBlackboard : public AlicaTestFixture
         targetBlueprint.addKey("targetKey", targetTypeName);
         Blackboard srcBB(&srcBlueprint);
         Blackboard targetBB(&targetBlueprint);
-        srcBB._impl.set("srcKey", srcValue);
-        targetBB._impl.map("srcKey", "targetKey", srcBB._impl);
+        srcBB.impl().set("srcKey", srcValue);
+        targetBB.impl().map("srcKey", "targetKey", srcBB.impl());
         if constexpr (std::is_same_v<TargetType, std::any>) {
-            return std::any_cast<SrcType>(targetBB._impl.get<TargetType>("targetKey")) == std::any_cast<SrcType>(targetValue);
+            return std::any_cast<SrcType>(targetBB.impl().get<TargetType>("targetKey")) == std::any_cast<SrcType>(targetValue);
         } else {
-            return targetBB._impl.get<TargetType>("targetKey") == targetValue;
+            return targetBB.impl().get<TargetType>("targetKey") == targetValue;
         }
     }
 
-    alica::internal::BlackboardImpl& getBlackboardImpl(alica::Blackboard& bb) { return bb._impl; }
+    alica::internal::BlackboardImpl& getBlackboardImpl(alica::Blackboard& bb) { return bb.impl(); }
 };
 
+TEST_F(SingleAgentTestFixture, testValueMappingBehaviours)
+{
+    // Checks if value mapping for behaviours succeeds
+
+    // Transition to the plan corresponding to this test case
+    ASSERT_TRUE(_tc->setTransitionCond("TestMasterPlan", "ChooseTestState", "BlackboardTestState")) << _tc->getLastFailure();
+    STEP_UNTIL(_tc, _tc->getActivePlan("BlackboardTestPlan"));
+    auto plan = _tc->getActivePlan("BlackboardTestPlan");
+    ASSERT_NE(plan, nullptr) << _tc->getLastFailure();
+
+    ASSERT_TRUE(_tc->setTransitionCond("BlackboardTestPlan", "ChooseBlackboardTestState", "ValueMappingBehaviourTestState")) << _tc->getLastFailure();
+    plan = _tc->getActivePlan("BlackboardTestPlan");
```

Comment:
need to check if the behaviour is active

Reply:
When we transition into the behaviour, the behaviours init will be called and the transition into the success state will be triggered immediately. I think we need additional flags that delay the transition out of the behaviour state for that check.